### ISSUE TITLE
Windows build

### ISSUE THREAD

**Issue by snej** [couchbaselabs/crouton#2]
Everything _should_ work on Windows, but I don't have a Windows system nor know anything about developing on that platform.

**Comment by snej** [comment]
I turned on Windows builds in CI and fixed a bunch of issues; now as of commit 9e3a5154 the only(?) error left is this doozy:

```
D:\a\crouton\crouton\src\HTTPHandler.cc(34): fatal  error C1001: Internal compiler error. [D:\a\crouton\crouton\build_cmake\debug\LibCrouton.vcxproj]
  (compiler file 'D:\a\_work\1\s\src\vctools\Compiler\Utc\src\p2\main.c', line 224)
```
https://github.com/couchbaselabs/crouton/actions/runs/6115096815/job/16598009674#step:4:438

**Comment by snej** [comment]
I was able to work around the compiler error by breaking the function down and simplifying some of the coroutine calls. Commit 08235a4 finally did the trick.

As of the next commit, d5edd69426959c7b30e0c5a14cd1c62a73790c3f, the build is green at last.
